tests/duplicate_event/run.sh

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+set -eu
+
+cur=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+source $cur/../_utils/test_prepare
+WORK_DIR=$TEST_DIR/$TEST_NAME
+TASK_NAME="test"
+SQL_RESULT_FILE="$TEST_DIR/sql_res.$TEST_NAME.txt"
+
+API_VERSION="v1alpha1"
+
+function run() {
+	# 1. test sync fetch binlog met error and reset binlog streamer with remote binlog
+
+	# with a 5 rows insert txn: 1 * GTID + 1 * BEGIN + 5 * (Table_map + Write_rows) + 1 * XID
+	# here we fail at the third write rows event, sync should retry and auto recover without any duplicate event
+	export GO_FAILPOINTS="github.com/pingcap/dm/syncer/GetEventErrorInTxn=13*return(3)"
+
+	run_sql_file $cur/data/db1.prepare.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
+	check_contains 'Query OK, 2 rows affected'
+
+	run_dm_master $WORK_DIR/master $MASTER_PORT $cur/conf/dm-master.toml
+	check_rpc_alive $cur/../bin/check_master_online 127.0.0.1:$MASTER_PORT
+	check_metric $MASTER_PORT 'start_leader_counter' 3 0 2
+	run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+	check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+
+	cp $cur/conf/source1.yaml $WORK_DIR/source1.yaml
+	dmctl_operate_source create $WORK_DIR/source1.yaml $SOURCE_ID1
```

Comment:
```suggestion
	dmctl_operate_source create $cur/conf/source1.yaml $SOURCE_ID1
```

if we don't modify $cur/conf/source1.yaml

and for source2